examples/trivial.rs

```diff
@@ -1,16 +1,29 @@
 use memoize::memoize;
 
 #[memoize]
```

Comment:
nice example, but given that this one is the `trivial` one, maybe keep it trivial? And instead set up a new example, e.g. `result.rs`?

Reply:
Done, i figured that results was also a trivial case, but thats fair enough.